install_test/install_test.cpp

```diff
@@ -0,0 +1,312 @@
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/ginkgo.hpp>
+
+
+#include <chrono>
+#include <iostream>
+#include <map>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+
+// core/base/polymorphic_object.hpp
+class PolymorphicObjectTest : public gko::PolymorphicObject {};
+
+
+int main(int, char **)
+{
+    auto refExec = gko::ReferenceExecutor::create();
+    // core/base/abstract_factory.hpp
+    {
+        using type1 = int;
+        using type2 = double;
+        static_assert(std::is_same<
+                      gko::AbstractFactory<type1, type2>::abstract_product_type,
+                      type1>::value);
+    }
+
+    // core/base/array.hpp
+    {
+        using type1 = int;
+        using ArrayType = gko::Array<type1>;
+        ArrayType test;
+    }
+
+    // core/base/combination.hpp
+    {
+        using type1 = int;
+        static_assert(
+            std::is_same<gko::Combination<type1>::value_type, type1>::value);
+    }
+
+    // core/base/composition.hpp
+    {
+        using type1 = int;
+        static_assert(
+            std::is_same<gko::Composition<type1>::value_type, type1>::value);
+    }
+
+    // core/base/dim.hpp
+    {
+        using type1 = int;
+        auto test = gko::dim<3, type1>{4, 4, 4};
+    }
+
+    // core/base/exception.hpp
+    {
+        auto test = gko::Error(std::string("file"), 12,
+                               std::string("Test for an error class."));
+    }
+
+    // core/base/exception_helpers.hpp
+    {
+        auto test = gko::dim<2>{3};
+        ASSERT_IS_SQUARE_MATRIX(test);
+    }
+
+    // core/base/executor.hpp
+    {
+        auto test = gko::ReferenceExecutor::create();
+    }
+
+    // core/base/math.hpp
+    {
+        using testType = double;
+        static_assert(gko::is_complex<testType>() == false);
+    }
+
+    // core/base/matrix_data.hpp
+    {
+        gko::matrix_data<> test{};
+    }
+
+    // core/base/mtx_io.hpp
+    {
+        auto test = gko::layout_type::array;
+    }
+
+    // core/base/name_demangling.hpp
+    {
+        auto testVar = 3.0;
+        auto test = gko::name_demangling::get_static_type(testVar);
+    }
+
+
+    // core/base/polymorphic_object.hpp
+    {
+        auto test = gko::layout_type::array;
+    }
+
+    // core/base/range.hpp
+    {
+        auto test = gko::span{12};
+    }
+
+    // core/base/range_accessors.hpp
+    {
+        auto testVar = 12;
+        auto test = gko::range<gko::accessor::row_major<decltype(testVar), 2>>(
+            &testVar, 1, 1, 1);
```

Comment:
This one causes the warning.

Reply:
```suggestion
            &testVar, 1u, 1u, 1u);
```

Should do the trick.

Reply:
Ah, I thought you meant the unused variables from the main function. Fixed it in the latest commit.